integration-tests/deployment/clo/don_nodeset.go

```diff
@@ -0,0 +1,65 @@
+package clo
+
+import (
+	"strings"
+
+	"github.com/smartcontractkit/chainlink/integration-tests/deployment/clo/models"
+)
+
+func DonNodesets(nops []*models.NodeOperator, donFilters map[string]FilterFuncT[*models.Node]) map[string][]*models.NodeOperator {
+	// first drop bootstraps if they exist
+	nonBootstrapNops := filterNopNodes(nops, func(n *models.Node) bool {
+		for _, chain := range n.ChainConfigs {
+			if chain.Ocr2Config.IsBootstrap {
+				return false
+			}
+		}
+		return true
+	})
+	out := make(map[string][]*models.NodeOperator)
+	for name, f := range donFilters {
+		out[name] = filterNopNodes(nonBootstrapNops, f)
```

Comment:
Did you mean to compose all filters? In that case shouldn't you be using the output of previous iteration instead of nonBootstrapNops every time?

Reply:
no, this is partitioning nodes into dons.

add some comments and renamed the func, hopefully clearer